
# HG changeset patch
# User Hans de Goede <hdegoede@redhat.com>
# Date 1265027566 -3600
# Node ID f23c5a878fb1581d2be9c19975307c2c8d463b61
# Parent  bf7af07e9bbdeb453b866829107d88496bf9f329
libv4l: Skip unknown huffman codes in sonixb decompression

From: Hans de Goede <hdegoede@redhat.com>

This slightly improves the image when unknown codes are encountered.

Priority: normal

Signed-off-by: Hans de Goede <hdegoede@redhat.com>

--- a/v4l2-apps/libv4l/libv4lconvert/sn9c10x.c	Sun Jan 31 17:25:32 2010 +0100
+++ b/v4l2-apps/libv4l/libv4lconvert/sn9c10x.c	Mon Feb 01 13:32:46 2010 +0100
@@ -38,9 +38,6 @@
 static code_table_t table[256];
 static int init_done = 0;
 
-/* global variable */
-static int sonix_unknown = 0;
-
 /*
 	sonix_decompress_init
 	=====================
@@ -113,7 +110,6 @@
 		table[i].unk = unk;
 	}
 
-	sonix_unknown = 0;
 	init_done = 1;
 }
 
@@ -172,8 +168,10 @@
 			/* update bit position */
 			bitpos += table[code].len;
 
-			/* update code statistics */
-			sonix_unknown += table[code].unk;
+			/* Skip unknown codes (most likely they indicate
+			   a change of the delta's the various codes encode) */
+			if (table[code].unk)
+				continue;
 
 			/* calculate pixel value */
 			val = table[code].val;

